### ISSUE TITLE
Test failure GC\\API\\Frozen\\Frozen\\Frozen.cmd

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#66569]
Run: [runtime-coreclr gcstress-extra 20220312.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1660529&view=ms.vss-test-web.build-test-results-tab&runId=45714724&resultId=108099&paneView=debug)

Failed test:
```
CoreCLR windows arm64 Checked heapverify1 @ Windows.10.Arm64v8.Open

- GC\\API\\Frozen\\Frozen\\Frozen.cmd

CoreCLR Linux arm64 Checked heapverify1 @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-20210531091519-97d8652

- GC/API/Frozen/Frozen/Frozen.sh

CoreCLR Linux x64 Checked heapverify1 @ Ubuntu.1804.Amd64.Open

- GC/API/Frozen/Frozen/Frozen.sh

```
**Error message:**
```
Assert failure(PID 7288 [0x00001c78], Thread: 336 [0x0150]): Consistency check failed: AV in clr at this callstack:
CORECLR! WKS::gc_heap::verify_heap + 0x76C (0x00007ffd7b39ebd4) CORECLR! WKS::gc_heap::gc1 + 0xAFC (0x00007ffd7b38375c)
CORECLR! WKS::gc_heap::garbage_collect + 0x6C0 (0x00007ffd7b382c20) CORECLR! WKS::GCHeap::GarbageCollectGeneration + 0x584 (0x00007ffd7b36b4cc)
CORECLR! WKS::GCHeap::GarbageCollect + 0x1B8 (0x00007ffd7b36aec8) CORECLR! GCInterface_Collect + 0x130 (0x00007ffd7b18c060)
<no module>! <no symbol> + 0x0 (0x00007ffd`1b98f950)
.AV on tid=0x150 (336), cxr=0000004873B7DB50, exr=0000004873B7DEE0
FAILED: false

CORECLR! CHECK::Trigger + 0x2D0 (0x00007ffd7b3b79b8) CORECLR! CLRVectoredExceptionHandlerPhase3 + 0x2F4 (0x00007ffd7b0563bc)
CORECLR! CLRVectoredExceptionHandlerPhase2 + 0x64 (0x00007ffd7b055e54) CORECLR! CLRVectoredExceptionHandler + 0x16C (0x00007ffd7b055dc4)
CORECLR! CLRVectoredExceptionHandlerShim + 0x24C (0x00007ffd7b05670c) NTDLL! RtlGetSaclSecurityDescriptor + 0x180 (0x00007ffdae7ac700)
NTDLL! LdrControlFlowGuardEnforced + 0x6D4 (0x00007ffdae7e5df4) NTDLL! KiUserExceptionDispatcher + 0x24 (0x00007ffdae783434)
CORECLR! WKS::gc_heap::verify_heap + 0x76C (0x00007ffd7b39ebd4) CORECLR! WKS::gc_heap::gc1 + 0xAFC (0x00007ffd7b38375c)
File: D:\a_work\1\s\src\coreclr\vm\excep.cpp Line: 7581
Image: D:\h\w\A9730951\p\corerun.exe

Return code:      1
Raw output file:      D:\h\w\A9730951\w\A3B0091F\uploads\Reports\GC.API\Frozen\Frozen\Frozen.output.txt
Raw output:
BEGIN EXECUTION
"D:\h\w\A9730951\p\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false"  Frozen.dll
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=D:\h\w\A9730951\p
D:\h\w\A9730951\w\A3B0091F\e\GC\API\Frozen\Frozen\Frozen.cmd
Expected: True
Actual:   False


Stack trace
   at GC_API._Frozen_Frozen_Frozen_._Frozen_Frozen_Frozen_cmd()
```

**Comment by VincentBu** [comment]
Failed again in: [runtime-coreclr gcstress-extra 20220326.1
](https://dev.azure.com/dnceng/public/_build/results?buildId=1683920&view=ms.vss-test-web.build-test-results-tab&runId=46090662&resultId=108083&paneView=debug)

Failed test:
```
CoreCLR windows arm64 Checked heapverify1 @ Windows.10.Arm64v8.Open

- GC\\API\\Frozen\\Frozen\\Frozen.cmd

CoreCLR windows x64 Checked heapverify1 @ Windows.10.Amd64.Open

- GC\\API\\Frozen\\Frozen\\Frozen.cmd

```
**Error message:**
```
Assert failure(PID 7612 [0x00001dbc], Thread: 16644 [0x4104]): Consistency check failed: AV in clr at this callstack:
CORECLR! WKS::gc_heap::verify_heap + 0x76C (0x00007ffcfaeff5f4) CORECLR! WKS::gc_heap::gc1 + 0xAFC (0x00007ffcfaee3f6c)
CORECLR! WKS::gc_heap::garbage_collect + 0x6C0 (0x00007ffcfaee3430) CORECLR! WKS::GCHeap::GarbageCollectGeneration + 0x584 (0x00007ffcfaecbdec)
CORECLR! WKS::GCHeap::GarbageCollect + 0x1B8 (0x00007ffcfaecb7e8) CORECLR! GCInterface_Collect + 0x130 (0x00007ffcfacec5b0)
<no module>! <no symbol> + 0x0 (0x00007ffc`9b5e0f58)
.AV on tid=0x4104 (16644), cxr=00000017A797E250, exr=00000017A797E5E0
FAILED: false

CORECLR! CHECK::Trigger + 0x2D0 (0x00007ffcfaf183d8) CORECLR! CLRVectoredExceptionHandlerPhase3 + 0x2F4 (0x00007ffcfabb63bc)
CORECLR! CLRVectoredExceptionHandlerPhase2 + 0x64 (0x00007ffcfabb5e54) CORECLR! CLRVectoredExceptionHandler + 0x16C (0x00007ffcfabb5dc4)
CORECLR! CLRVectoredExceptionHandlerShim + 0x24C (0x00007ffcfabb670c) NTDLL! RtlGetSaclSecurityDescriptor + 0x180 (0x00007ffd2c82ccb0)
NTDLL! LdrControlFlowGuardEnforced + 0x794 (0x00007ffd2c7ecdc4) NTDLL! KiUserExceptionDispatcher + 0x24 (0x00007ffd2c7c3434)
CORECLR! WKS::gc_heap::verify_heap + 0x76C (0x00007ffcfaeff5f4) CORECLR! WKS::gc_heap::gc1 + 0xAFC (0x00007ffcfaee3f6c)
File: D:\a_work\1\s\src\coreclr\vm\excep.cpp Line: 7581
Image: D:\h\w\AFCB08F3\p\corerun.exe

Return code:      1
Raw output file:      D:\h\w\AFCB08F3\w\AC560905\uploads\Reports\GC.API\Frozen\Frozen\Frozen.output.txt
Raw output:
BEGIN EXECUTION
"D:\h\w\AFCB08F3\p\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false"  Frozen.dll
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=D:\h\w\AFCB08F3\p
D:\h\w\AFCB08F3\w\AC560905\e\GC\API\Frozen\Frozen\Frozen.cmd
Expected: True
Actual:   False


Stack trace
   at GC_API._Frozen_Frozen_Frozen_._Frozen_Frozen_Frozen_cmd()
```

**Comment by MichalStrehovsky** [comment]
@mangod9 this started failing after enabling regions. It might indicate that regions still don't work with frozen segments. The root cause could be the same with NativeAOT.

**Comment by mangod9** [comment]
@cshung as fyi. We will look into this soon.

**Comment by VincentBu** [comment]
Failed again in: [runtime-coreclr gcstress-extra 20220405.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1701692&view=ms.vss-test-web.build-test-results-tab&runId=46383894&resultId=107912&paneView=debug)

Failed test:
```
CoreCLR windows arm64 Checked heapverify1 @ Windows.10.Arm64v8.Open

- GC\\API\\Frozen\\Frozen\\Frozen.cmd

CoreCLR Linux arm64 Checked heapverify1 @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-20210531091519-97d8652

- GC/API/Frozen/Frozen/Frozen.sh

CoreCLR Linux x64 Checked heapverify1 @ Ubuntu.1804.Amd64.Open

- GC/API/Frozen/Frozen/Frozen.sh

CoreCLR windows x64 Checked heapverify1 @ Windows.10.Amd64.Open

- GC\\API\\Frozen\\Frozen\\Frozen.cmd
```
**Error message:**
```
Assert failure(PID 6024 [0x00001788], Thread: 10620 [0x297c]): Consistency check failed: AV in clr at this callstack:
CORECLR! WKS::gc_heap::verify_heap + 0x76C (0x00007fffb7bff53c) CORECLR! WKS::gc_heap::gc1 + 0xAFC (0x00007fffb7be3ddc)
CORECLR! WKS::gc_heap::garbage_collect + 0x6C0 (0x00007fffb7be32a0) CORECLR! WKS::GCHeap::GarbageCollectGeneration + 0x584 (0x00007fffb7bcbcbc)
CORECLR! WKS::GCHeap::GarbageCollect + 0x1B8 (0x00007fffb7bcb6b8) CORECLR! GCInterface_Collect + 0x130 (0x00007fffb79ec5e0)
<no module>! <no symbol> + 0x0 (0x00007fff`582f0f58)
.AV on tid=0x297c (10620), cxr=0000000D3B57E1F0, exr=0000000D3B57E580
FAILED: false

CORECLR! CHECK::Trigger + 0x2D0 (0x00007fffb7c18318) CORECLR! CLRVectoredExceptionHandlerPhase3 + 0x2F4 (0x00007fffb78b63bc)
CORECLR! CLRVectoredExceptionHandlerPhase2 + 0x64 (0x00007fffb78b5e54) CORECLR! CLRVectoredExceptionHandler + 0x16C (0x00007fffb78b5dc4)
CORECLR! CLRVectoredExceptionHandlerShim + 0x24C (0x00007fffb78b670c) NTDLL! RtlGetSaclSecurityDescriptor + 0x180 (0x00007fffe982ccb0)
NTDLL! LdrControlFlowGuardEnforced + 0x794 (0x00007fffe97ecdc4) NTDLL! KiUserExceptionDispatcher + 0x24 (0x00007fffe97c3434)
CORECLR! WKS::gc_heap::verify_heap + 0x76C (0x00007fffb7bff53c) CORECLR! WKS::gc_heap::gc1 + 0xAFC (0x00007fffb7be3ddc)
File: D:\a_work\1\s\src\coreclr\vm\excep.cpp Line: 7581
Image: D:\h\w\C1BB0A34\p\corerun.exe

Return code:      1
Raw output file:      D:\h\w\C1BB0A34\w\BCEB09B3\uploads\Reports\GC.API\Frozen\Frozen\Frozen.output.txt
Raw output:
BEGIN EXECUTION
"D:\h\w\C1BB0A34\p\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false"  Frozen.dll
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=D:\h\w\C1BB0A34\p
D:\h\w\C1BB0A34\w\BCEB09B3\e\GC\API\Frozen\Frozen\Frozen.cmd
Expected: True
Actual:   False


Stack trace
   at GC_API._Frozen_Frozen_Frozen_._Frozen_Frozen_Frozen_cmd()
```

**Comment by Maoni0** [comment]
I don't think there's anything new here - it's just when @cshung checked in this test, it was never run with heap verification. we just need to check to filter out the object if it's in an ro region when we check if we need a card for it.

**Comment by MichalStrehovsky** [comment]
I think @Maoni0 fixed this in #67729 per the comment in the PR. Closing.
